### PR TITLE
feat: Add integration test to verify timer cleanup on connection close events

### PR DESCRIPTION
**Pull request by Kofacts**
This PR adds an integration test to verify that timers set by the middleware are properly cleaned up when a client connection is closed before the delay completes.

This ensures there are no memory leaks or unintended callback executions in such scenarios, improving the reliability and robustness of the package.

The test covers the edge case where a connection is closed before the delay timer fires, and asserts that the timer is cleared and the next middleware is not called.

### CODE REVIEW

File: source/slow-down.ts [hunk 1]
```diff
@@ -94,7 +94,7 @@ export const slowDown = (
 			? { default: notUndefinedOptions.validate }
 			: notUndefinedOptions.validate ?? { default: true }
 
-	// TODO: Remove in v3.
+	// Deprecation warning for delayMs behavior change from v2
```
Comment (nfriedly) [inline source/slow-down.ts]: Oof, I should probably actually remove this check. I think it was important to have there when we first released v2, but I don't think it's needed at this point. I just forgot to remove it when we did the v3 bump.

Doesn't need to be part of this PR, though.

**Review by nfriedly — APPROVED** [review]
Hey, thanks!

This seems a like a good test (and other fixes 🤦 ) - although I'm curious why you decided to add this test. Did you see a case where the library wasn't working correctly?

Now that express-slow-down is built on top of express-rate-limit, the former benefits from all of the latter's tests, and I think there's one to cover this scenario. But, it doesn't hurt to have one here also.
